### DISCUSSION TITLE
Person-Card component no longer works within layouts > People > Manage people fields

### DISCUSSION THREAD

**Discussion by Justin-Nash** [microsoft-search/pnp-modern-search#4167] [Q&A]
**Version used**
4.12.2

**Describe the bug**
In older versions of PnP Modern Search we were able to use Microsoft Graph [Person-Card component](https://learn.microsoft.com/en-us/graph/toolkit/components/person-card?tabs=html) to render a manager display name based on the managers email address returned in search after configuring a managed property Manager. 

This was working well allowing users to see both the user information on the person they where looking for and that persons manager without having to click through pop ups.

The issue now is that the Microsoft Graph [Person-Card component](https://learn.microsoft.com/en-us/graph/toolkit/components/person-card?tabs=html) no longer works within the "Manage people fields" section. The last known working version was 4.9. We haven't tested ever version between 4.9 and 4.12.2 but I know the issue exists in:

-  4.12.2
-  4.12.1
-  4.11.1

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Add a new ***PnP - Search Results*** web part to a page
2. Select ***SharePoint Search***
3. Set Result Source Id/ Scope|Name to ***LocalPeopleResults***
4. Select ***Next*** so you are on page 2
5. Select ***People*** under Handlebars/HTML
6. Select ***Managed card fields***
7. Select the edit icon for ***Primary text**
8. Replace `{{slot item @root.slots.UserDisplayName}}` with `<mgt-person user-id="{{getUserEmail (slot item @root.slots.UserEmail)}}" view="oneline"></mgt-person>`
9. Save, Save and Publish

**Expected behavior**
We expect to see the Users Name in the first field. As demonstrated in the screenshots.

**Screenshots**
Expected outcome:
![image](https://github.com/user-attachments/assets/031308e6-347e-4253-aeb3-4f23952c310b)

Actual outcome:
![image](https://github.com/user-attachments/assets/7d6484af-e8ff-41c2-8070-68b8311d584e)

**Desktop (please complete the following information):**
 - Browser Chrome, Edge, Firefox

**Additional context**
The above steps are a simplified version of what we do with a Manager field. For our environment using a users email or UPN is not a valid option as they do not include the users name in full or in any consistent order.

**Comment by kasperbolarsen** [comment]
HI, you are right, that doesn't seem to work 

Perhaps a custom template like this can resolve the issue:
![image](https://github.com/user-attachments/assets/c99e75ed-82d2-495e-a670-f6dcf4a4050a)


In order for it to work you will have to set a slot to "Manager" and in my case the manager managed property was mapped to a RefinableString , but I guess you can use the auto generated one as well. Remember to add it to "selected properties" as well


<content id="data-content">

    <style>
        mgt-person {
            --initials-background-color: {{@root.theme.palette.themePrimary}};
        }

    </style>

    <div class="template">

        {{#if @root.properties.showSelectedFilters}}
            <pnp-selectedfilters 
                data-filters="{{JSONstringify filters.selectedFilters 2}}" 
                data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
                data-instance-id="{{filters.instanceId}}" 
                data-operator="{{filters.filterOperator}}"
                data-theme-variant="{{JSONstringify @root.theme}}"
            >
            </pnp-selectedfilters>
        {{/if}}
       
        <div class="template--header">
            {{#if @root.properties.showResultsCount}}
                <div class="template--resultCount">
                    <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
                </div>
            {{/if}}

            <div class="template--sort">
                <pnp-sortfield 
                    data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}" 
                    data-default-selected-field="{{sort.selectedSortFieldName}}" 
                    data-default-direction="{{sort.selectedSortDirection}}"
                    data-theme-variant="{{JSONstringify @root.theme}}">
                </pnp-sortfield>    
            </div>
        </div>
        
        <div>
            <ul class="template--defaultList">
                {{#each data.items as |item|}}
                    <pnp-select 
                        data-enabled="{{@root.properties.itemSelectionProps.allowItemSelection}}" 
                        data-index="{{@index}}" 
                        data-is-selected="{{isItemSelected @root.selectedKeys @index}}">

                        <template id="content">
                            <li class="template--peopleListItem" tabindex="0">
                                {{#> resultTypes item=item}}

                                    {{#and @root.properties.layoutProperties.showPersonaCard (slot item @root.slots.PersonQuery)}}
                                        <mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">
                                            
                                            <template data-type="person-card">
                                                <mgt-person-card inherit-details>
                                                </mgt-person-card>
                                            </template>
                                        </mgt-person>
                                    {{else}}
                                          
                                    <div>
                                        
                                        <mgt-person person-query="{{getUserEmail (slot item @root.slots.UserEmail)}}" view="threelines"></mgt-person>
                                          <div>Manager</div>
                                          <ul role="menu">
                                            <li role="menuitem" data-is-focusable="true" class="ui-list__item" tabindex="0">
                                              <mgt-person person-query="{{#with (split (slot item @root.slots.Manager) '|')}}{{[2]}}{{/with}}" view="twolines" line2-property="jobTitle"></mgt-person>
                                            </li>
                                            
                                          </ul>
                                      </div>
                                      

                                    {{/and}}
                                    
                                {{/resultTypes}}
                            </li>
                        </template>
                            
                    </pnp-select>
                {{/each}}
            </ul>
        </div>

        {{#if @root.properties.paging.showPaging}}

            {{#gt @root.data.totalItemsCount @root.properties.paging.itemsCountPerPage}}
                <pnp-pagination 
                    data-total-items="{{@root.data.totalItemsCount}}" 
                    data-hide-first-last-pages="{{@root.properties.paging.hideFirstLastPages}}"
                    data-hide-disabled="{{@root.properties.paging.hideDisabled}}"
                    data-hide-navigation="{{@root.properties.paging.hideNavigation}}"
                    data-range="{{@root.properties.paging.pagingRange}}" 
                    data-items-count-per-page="{{@root.properties.paging.itemsCountPerPage}}" 
                    data-current-page-number="{{@root.paging.currentPageNumber}}"
                    data-theme-variant="{{JSONstringify @root.theme}}"
                >
                </pnp-pagination>
            {{/gt}}
            
        {{/if}}

    </div>

</content>

<content id="placeholder-content">
    <div class="placeholder">

        {{#if @root.properties.showResultsCount}}
            <div class="template--resultCount">
                <span class="placeholder--shimmer placeholder--line" style="width: 20%"></span>
            </div>
        {{/if}}

        <div>
            <ul class="template--defaultList">
                {{#times @root.properties.paging.itemsCountPerPage}}   
                    <pnp-personashimmers 
                        data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                        data-theme-variant="{{JSONstringify @root.theme}}"
                    >
                    </pnp-personashimmers>
                {{/times}}
            </ul>
        </div>
    </div>
</content>

**Comment by itebet** [comment]
It doesn't work neither on my side.
On previous versions that code worked as a charm, but now it's just showing nothing (adaptive cards custom template):
   {
                    "type": "TextBlock",
                    "text": "<mgt-person person-card='click' avatar-size='large' view='oneline' user-id='${split($data.AuthorOWSUSER, ' | ')[0]}'/>",
                    "color": "Accent",
                    "size": "Medium",
                    "maxLines": 3
                }

**Comment by patrikhellgren** [comment]
Graph toolkit was upgraded to version 4 in version 4.11 of PnP Modern Search and now uses [disambiguation](https://learn.microsoft.com/en-us/graph/toolkit/customize-components/disambiguation) to not interfere with other versions of mgt loaded on the page. All components of mgt running in Modern Search now has a disambiguation string of _pnp-modern-search_ so mgt-person now becomes mgt-pnp-modern-search-person when loaded on the page by Modern Search. This should be handled behind the scenes but seems to be missing in some places so probably a bug (I'll try to have a look). While waiting for a fix you can use the workaround to change it yourself in your configurations to something like the following:

```html
<mgt-pnp-modern-search-person user-id="{{getUserEmail (slot item @root.slots.UserEmail)}}" view="oneline"></mgt-pnp-modern-search-person>
```

```json
{
"type": "TextBlock",
"text": "<mgt-pnp-modern-search-person person-card='click' avatar-size='large' view='oneline' user-id='${split($data.AuthorOWSUSER, ' | ')[0]}'/>",
"color": "Accent",
"size": "Medium",
"maxLines": 3
}
```

The workaround should probably keep working even if and when the automatic disambiguation starts working.

**EDIT** this has now been addressed in #4169

**Comment by wobba** [comment]
Fixed in the dev branch with #4169 4169
